### PR TITLE
Fix error labels when missing LTC configuration

### LINKED ISSUE #915: missing parameters are mis-labeled for Litecoin in error
When creating a lnd.conf if the following parameters are missing or left blank

```
litecoind.rpcuser=kek
litecoind.rpcpass=kek
litecoind.zmg
```

or the corresponding `ltcd` params then the error indicates that the `bitcoind` or `btcd` parameters are missing, which is of course incorrect.

### PR DESCRIPTION
**Pull request by PWKad**
Fix #915 by capturing the current daemonName and using that in
error messages.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -805,20 +805,27 @@ func parseRPCParams(cConfig *chainConfig, nodeConfig interface{}, net chainCode,
...
+		case litecoinChain:
+			daemonName = "ltcd"
+		}
+
 		// If only ONE of RPCUser or RPCPass is set, we assume the
 		// user did that unintentionally.
 		if conf.RPCUser != "" || conf.RPCPass != "" {
 			return fmt.Errorf("please set both or neither of " +
-				"btcd.rpcuser and btcd.rpcpass")
+				daemonName + ".rpcuser and " + daemonName + ".rpcpass")
```
Comment (cfromknecht) [inline config.go]: Since we’re already using `Errorf`, this can use formatting directives instead of string concatenation

File: config.go [hunk 2]
```diff
@@ -828,21 +835,27 @@ func parseRPCParams(cConfig *chainConfig, nodeConfig interface{}, net chainCode,
...
+			daemonName = "litecoind"
+		}
 		// If only one or two of the parameters are set, we assume the
 		// user did that unintentionally.
 		if conf.RPCUser != "" || conf.RPCPass != "" || conf.ZMQPath != "" {
 			return fmt.Errorf("please set all or none of " +
-				"bitcoind.rpcuser, bitcoind.rpcpass, and " +
-				"bitcoind.zmqpath")
+				daemonName + ".rpcuser, " + daemonName +  ".rpcpass, " +
+				"and " + daemonName + ".zmqpath")
```
Comment (cfromknecht) [inline config.go]: Same comment about formatting instead of string concat

File: config.go (hunk 1 above)
Comment (PWKad) [reply config.go]: I'm not super-familiar with go, going to update the commit like this - 

```
                        return fmt.Errorf("please set all or none of " +
                                "%[1]v.rpcuser, %[1]v.rpcpass, " +
                                "and %[1]v.zmqpath", daemonName)
```

Does that look correct?

**Comment by PWKad** [comment]
@cfromknecht  updated whenever you get a chance to have a look.

File: config.go [hunk 3]
```diff
@@ -861,20 +861,27 @@ func parseRPCParams(cConfig *chainConfig, nodeConfig interface{}, net chainCode,
 		if conf.RPCUser != "" && conf.RPCPass != "" {
 			return nil
 		}
+
+		// Get the daemon name for displaying proper errors
```
Comment (cfromknecht) [inline config.go]: nit: we try to use complete sentences in all of our comments, just a period would suffice

File: config.go [hunk 4]
```diff
@@ -884,21 +891,27 @@ func parseRPCParams(cConfig *chainConfig, nodeConfig interface{}, net chainCode,
 		if conf.RPCUser != "" && conf.RPCPass != "" && conf.ZMQPath != "" {
 			return nil
 		}
+
+		// Get the daemon name for displaying proper errors
```
Comment (cfromknecht) [inline config.go]: nit: same as above

**Comment by PWKad** [comment]
Ok thank for pointing that out, I must have missed that when reading over the contribution guidelines but will check them out again.

Updated the commit message and added the periods for the sentences.  Typically since this is a fork I've just been force pushing to keep one commit but that kind of messes with GitHubs code review features so if there is a better workflow just let me know.

**Comment by patrick-walters-q2** [comment]
Oh nice, no I think I have a pretty high knowledge of git but somehow have never seen or used that feature, will definitely keep that in mind.  Thanks for sharing.

**Pull request merged**